src/librbd/ImageCtx.h

```diff
@@ -61,6 +62,43 @@ namespace librbd {
   template <typename> class ResizeRequest;
   }
 
+  class ImageQosStatusSingleton {
```

Comment:
@dillaman, the iops limit is per image, status should be globally shared by all users of that image, so I make a singleton, does it make sense?